osf/management/commands/sync_citation_styles.py

```diff
@@ -0,0 +1,135 @@
+                    'has_bibliography': has_bibliography, 'parent_style': None,
+                    'short_title': short_title,
+                    'summary': summary
+                }
+
+                CitationStyle.objects.update_or_create(**fields)
+
+            for file_name in [name for name in zip_file.namelist() if name.endswith('.zip') and name.startswith('dependent')]:
+                root = etree.fromstring(zip_file.read(file_name))
+
+                namespace = re.search(r'\{.*\}', root.tag).group()
+                title = root.find('{namespace}info/{namespace}title'.format(namespace=f'{namespace}')).text
+                has_bibliography = root.find(f'{{{namespace}}}bibliography') is not None or 'Bluebook' in title
+
+                style_id = os.path.splitext(os.path.basename(file_name))[0]
+                links = root.findall('{namespace}info/{namespace}link'.format(namespace=f'{namespace}'))
+                for link in links:
+                    if link.get('rel') == 'independent-parent':
+                        parent_style_id = urlparse(link.get('href')).path.split('/')[-1]
+                        parent_style = CitationStyle.objects.get(_id=parent_style_id)
+
+                        if parent_style is not None:
+                            summary = getattr(
+                                root.find('{namespace}info/{namespace}summary'.format(namespace=f'{namespace}')),
+                                'text', None
+                            )
+                            short_title = getattr(
+                                root.find('{namespace}info/{namespace}title-short'.format(namespace=f'{namespace}')),
+                                'text', None
+                            )
+
+                            parent_has_bibliography = parent_style.has_bibliography
+                            fields = {
+                                '_id': style_id,
+                                'title': title,
+                                'has_bibliography': parent_has_bibliography,
+                                'parent_style': parent_style_id,
+                                'short_title': short_title,
+                                'summary': summary
+                            }
+                            CitationStyle.objects.update_or_create(**fields)
+                            break
+                        else:
+                            logger.debug('Unable to load parent_style object: parent {}, dependent style {}'.format(parent_style_id, style_id))
+                else:
+                    fields = {
+                        '_id': style_id,
+                        'title': title,
+                        'has_bibliography': has_bibliography,
+                        'parent_style': None
+                    }
+                    CitationStyle.objects.update_or_create(**fields)
+
+        if dry_run:
+            raise RuntimeError('This is a dry run rolling back transaction.')
```

Comment:
Minor: is this the normal way how we implement dry-run for Django commands (i.e. raise an exception to roll back transaction)?

Reply:
There is no normal way to do a dry_run, but this was the same strategy used in `fix_quickfiles_waterbutler_logs.py` so yes it's pretty standard when one transaction is used.